### ISSUE TITLE
Harmonize disease type separators

### ISSUE THREAD

**Issue by jashapiro** [AlexsLemonade/OpenPBTA-analysis#222]
#### File(s)

pbta-histologies.tsv

#### Release

v7

#### Question/issue

The meanings of separators (such as`;`, `/` and `-`) in the disease_type_new column are unclear, and possibly inconsistent. 

As noted in #213, semicolons sometimes seem to refer to subtypes and/or additional information, as in the case of `Low-grade glioma;astrocytoma (WHO grade I/II)`, but at other times are used to separate what appears to be multiple reported pathologies, as in `Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II)` Note that in this second example, there is also a `/`, which now is being used where the first case used a semicolon.  ~In another case, `High-grade glioma;astrocytoma (WHO grade III/IV), Low-grade glioma;astrocytoma (WHO grade I/II)` a `, ` is used to separate multiple reported pathologies.~ [Edit: I was looking at the wrong table when I found this one]

In another case, `Brainstem glioma- Diffuse intrinsic pontine glioma`, a `-` is used to separate descriptors. 

I would suggest replacing the semicolon in the case of multiple types with a `, ` (including a space), and standardizing the use of `;` or `/` for subtypes or secondary descriptors.

**Comment by jashapiro** [comment]
Subquestion: In the `short_histology` column, both `High-grade glioma;astrocytoma (WHO grade III/IV)` and `Brainstem glioma- Diffuse intrinsic pontine glioma` are combined as `HGAT`. Is this what we want for analysis, or should the latter be listed as `DIPG`?

**Comment by jharenza** [comment]
> #### File(s)
> pbta-histologies.tsv
> 
> #### Release
> v7
> 
> #### Question/issue
> The meanings of separators (such as`;`, `/` and `-`) in the disease_type_new column are unclear, and possibly inconsistent.
> 
> As noted in #213, semicolons sometimes seem to refer to subtypes and/or additional information, as in the case of `Low-grade glioma;astrocytoma (WHO grade I/II)`, but at other times are used to separate what appears to be multiple reported pathologies, as in `Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II)` Note that in this second example, there is also a `/`, which now is being used where the first case used a semicolon. In another case, `High-grade glioma;astrocytoma (WHO grade III/IV), Low-grade glioma;astrocytoma (WHO grade I/II)` a `, ` is used to separate multiple reported pathologies.
> 
> In another case, `Brainstem glioma- Diffuse intrinsic pontine glioma`, a `-` is used to separate descriptors.
> 
> I would suggest replacing the semicolon in the case of multiple types with a `, ` (including a space), and standardizing the use of `;` or `/` for subtypes or secondary descriptors.

Regarding this, this is going to be low priority - this is something that goes back to the way the data was put into databases to not make our lives miserable when adding additional data, we should keep as much consistency with the KF DRC as possible so not to have to redo much of this later. Eg: we are supposed to be adding multiple additional PNOC samples and the nomenclature for disease type follows that `Brainstem glioma- Diffuse intrinsic pontine glioma` because it is in several databases that way. Changing now means making more of the same changes later.

**Comment by jharenza** [comment]
> Subquestion: In the `short_histology` column, both `High-grade glioma;astrocytoma (WHO grade III/IV)` and `Brainstem glioma- Diffuse intrinsic pontine glioma` are combined as `HGAT`. Is this what we want for analysis, or should the latter be listed as `DIPG`?

I believe I did this to abbreviate the `broad_histologies`, in which case HGAT describes both of these.

**Comment by jharenza** [comment]
We also cannot use `,` for these for technical reasons - the KF DRC feeds into CAVATICA and CAVATICA works on CSV files, so having these delimited by `,` messes up the columns. So, the DRC had tried to replace all commas with semicolons where possible.

**Comment by jharenza** [comment]
Tagging @allisonheath here

**Comment by jashapiro** [comment]
> > Subquestion: In the `short_histology` column, both `High-grade glioma;astrocytoma (WHO grade III/IV)` and `Brainstem glioma- Diffuse intrinsic pontine glioma` are combined as `HGAT`. Is this what we want for analysis, or should the latter be listed as `DIPG`?
> 
> I believe I did this to abbreviate the `broad_histologies`, in which case HGAT describes both of these.

Okay, that makes sense. Would you suggest that I keep these separate for downstream analyses or use the `broad_histology`? i.e. for mutation co-occurence data?

**Comment by jharenza** [comment]
I think the co-occurrance should probably be done on `disease_type_new`

**Comment by allisonheath** [comment]
Thanks for pointing this out, just as a bit of background, on the original collection forms here are the options provided (as checkboxes):

```
Medulloblastoma
Supratentorial or Spinal Cord PNET
Dysembryoplastic neuroepithelial tumor (DNET)
Ependymoma
Pineoblastoma
Craniopharyngioma
Low-grade glioma/astrocytoma (WHO grade I/II)
High-grade glioma/astrocytoma (WHO grade III/IV)
Ganglioglioma
Oligodendroglioma
Brainstem glioma- Diffuse intrinsic pontine glioma
Brainstem glioma- Tectal
Brainstem glioma- NOS
Metastatic secondary tumors
Dysplasia/Gliosis
Primary CNS lymphoma
Atypical Teratoid Rhabdoid Tumor (ATRT)
Neurofibroma/Plexiform
Malignant peripheral nerve sheath tumor (MPNST)
Germinoma
Non-germinomatous germ cell tumor
Teratoma
Choroid plexus papilloma
Choroid plexus carcinoma
Meningioma
Gliomatosis Cerebri
Neurocytoma
Langerhans Cell histiocytosis
Schwannoma
Chordoma
Adenoma
Glial-neuronal tumor NOS
Cavernoma
Ewing's Sarcoma
Ganglioneuroblastoma
Hemangioblastoma
Neuroblastoma
Rhabdomyosarcoma
Sarcoma
Subependymal Giant Cell Astrocytoma (SEGA)
Other
```
These forms were created quite a while ago and in the context of prompts for better manual record abstraction. We're currently working on revamping for the future, but for the current purposes, we'll have to make due with what we have. So as you can see, in the original collection, `- ` (note the space) was used to indicate specific subcategories (and `-` without a space for actual terms with dashes...), while `/` was used to indicate bucketed/equivalent diagnoses. When processing the collected data "as is" we use `;` as the standard separator to indicate multiple checked values. 

It looks like at some point when generating the `disease_type_new` the `/` got replaced by `;`, while really it looks like the intent was to clarify versus indicating two separate diagnoses. Will circle back with the team here on when we can update, but my initial suggestion for `disease_type_new` would be either 1) go ahead and keep them "as is" 2) if you want to streamline it we should document those transforms and I think they should be one-to-one. For example one suggestion could be:

`Low-grade glioma/astrocytoma (WHO grade I/II)` -> `Low-grade glioma`
`High-grade glioma/astrocytoma (WHO grade III/IV)` -> `High-grade glioma`
`Neurofibroma/Plexiform` -> `Neurofibroma`
`Dysplasia/Gliosis` -> `Dysplasia`
And then also removing the abbreviations in parentheses contained after some of the diagnoses.

Does this make sense? Opinions between the two options for downstream analysis purposes?

**Comment by jashapiro** [comment]
That is very helpful, thanks @allisonheath. The list of expected values is particularly useful. My guess is that the switch from `/` to `;` was somehow not done completely, perhaps partly intentionally. That would explain the fact that `WHO grade I/II` etc. was left as is, but that cases where two boxes might have been checked originally, i.e. `Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II)`, the `/` was not converted. Perhaps the simplest option to distinguish multiple tumor types would be to add a space after the `;` in cases like that? 

For downstream analysis, I do think a column of simple disease ids would be useful, akin to the `short_histology` column, but mapped 1to1, as you suggest, with an appropriate dictionary documenting the transform. My main suggestion for that would be that each be a single word, or `_` separated, to ease use in computational pipelines.

I don't think any of this is high priority, but it does have the potential to simplify code surrounding analysis of differences among diseases. This may also be helpful for simplifying and unifying plot labels down the line.

**Comment by jharenza** [comment]
@jashapiro I can try to work on this for V12 #326 per your and @allisonheath's suggestions

**Comment by jashapiro** [comment]
Looking at #347 to check on this, I noticed one sample where there is a change that is presumably the result of these adjustments, and I want to check that the fields are correct:

BS_K049HVGR and BS_ME3HKN2C are listed in `disease_type_old` as `Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II)` and now in `disease_type_new` as `Low-grade glioma`, but the `broad_histology` column is `Ganglioglioma`, whereas all other Low-grade glioma specimens are listed as `LGAT`. 

Another sample to check: 
BS_E0AAK0V0 seems to be the only `Dysplasia` | `DNET` pair. Again, this is a case where `disease_type_old` had many disease types: `Dysembryoplastic neuroepithelial tumor (DNET);Dysplasia/Gliosis;Ganglioglioma`

I found no other cases (though I may not have been exhaustive) where a `disease_type_new`  has more than one `broad_histology` associated with it, aside from DIPG and High Grade Glioma, which are sometimes listed as `HGAT` and sometimes `HGDAT` (I assume these are correct, given the frequency)

**Comment by jharenza** [comment]
@jashapiro thanks so much for catching these! I checked back to what was previously used for pedcbio characterization in the cases of >1 diagnosis and found the following, so changed these accordingly:

```
Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II) -> Low-grade mixed tumor -> LGMT
Dysembryoplastic neuroepithelial tumor (DNET);Dysplasia/Gliosis;Ganglioglioma -> Dysembryoplastic neuroepithelial tumor -> DNET
Dysembryoplastic neuroepithelial tumor (DNET);Ganglioglioma -> Dysembryoplastic neuroepithelial tumor -> 
Dysplasia/Gliosis;Glial-neuronal tumor NOS -> Low-grade mixed tumor -> LGMT
```

I also harmonized the `HGAT` - this must have been me using two different nomenclatures to distinguish diffusely infiltrating tumors, such as DIPGs, but looks like I missed that after some time, so for ease, we will just do `HGAT` for all of those as a broader category.

**Comment by jaclyn-taroni** [comment]
Closed via v12 release (#347)
